shell/browser/api/electron_api_web_contents.cc

```diff
@@ -535,7 +536,22 @@ void WebContents::InitWithSessionAndOptions(
   managed_web_contents()->GetView()->SetDelegate(this);
 
   auto* prefs = web_contents()->GetMutableRendererPrefs();
-  prefs->accept_languages = g_browser_process->GetApplicationLocale();
+
+  // Collect preferred languages from OS and browser process. accept_languages
+  // effects HTTP header, navigator.languages, and CJK fallback font selection.
+  //
+  // Note that an application locale set to the browser process might be
+  // different with the one set to the preference list.
+  // (e.g. overridden with --lang)
+  std::string accept_languages =
+      g_browser_process->GetApplicationLocale() + ",";
+  for (auto const& language : electron::GetPreferredLanguages()) {
+    if (language == g_browser_process->GetApplicationLocale())
```

Comment:
would there be issue with fallback if there was a duplicate in the final string ?

Reply:
Curious if we can remove this check incase its not necessary.

Reply:
confirmed `navigator.languages`, and `Accept-Language` header would contain duplicated entry; this should have no issue but look strange from end-users maybe?

Reply:
Cool, thanks for checking. In that case we can leave it in.